src/api/custom-resolvers/queries/get-application-by-reference-number/index.ts

```diff
@@ -0,0 +1,61 @@
+import { Application, Context } from '.keystone/types'; // eslint-disable-line
+import { getApplicationByReferenceNumberResponse, getApplicationByReferenceNumberVariables } from '../../../types';
+import getPopulatedApplication from '../../../helpers/get-populated-application';
+import decryptApplication from '../../../helpers/decrypt-application';
+
+/**
+ * getApplicationByReferenceNumber
+ * Gets an application by id
+ * Based on decrypt variables, decrypts part of application
+ * returns full application
+ * @param {Object} GraphQL root variables
+ * @param {Object} GraphQL variables for the getApplicationByReferenceNumberVariables mutation
+ * @param {Object} KeystoneJS context API
+ * @returns {Promise<getApplicationByReferenceNumberResponse>} Application with success flag
+ */
+const getApplicationByReferenceNumber = async (
+  root: any,
+  variables: getApplicationByReferenceNumberVariables,
+  context: Context,
+): Promise<getApplicationByReferenceNumberResponse> => {
+  try {
+    console.info('Getting application by reference number');
+
+    const { referenceNumber, decryptFinancialUk } = variables;
+
+    // finds applicationIds by id
+    const applicationIds = (await context.db.Application.findOne({
```

Comment:
I think this should be called just `application`, it's not touching application IDs and it's calling `.findOne`?